llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10033,12 +10040,40 @@ InstructionCost BoUpSLP::getTreeCost(ArrayRef<Value *> VectorizedVals) {
         }
       }
     }
+    // Leave the GEPs as is, they are free in most cases and better to keep them
+    // as GEPs.
+    TTI::TargetCostKind CostKind = TTI::TCK_RecipThroughput;
```

Comment:
why did you need to move this?

Reply:
To use it in TTI->getInstructionCost for adding an extra cost for the scalar GEP, which we're going to keep